samples/99_zil_interpreter/app/main.rb

```diff
@@ -7,39 +7,106 @@
 
 def tick(args)
   setup(args) if args.tick_count.zero?
-  # TODO:
-  # - Render history
-  # - Collect input
-  # - Call send_input with input when pressing Enter
+  # Render history
+  render_state(args)
+  
+  # Collect input
+  handle_input(args)
+  
+  # Call send_input when pressing Enter
+  if args.inputs.keyboard.key_down.enter && args.state.input != ''
+    send_input(args, args.state.input)
+    args.state.input = ''
```

Comment:
how about resetting `args.state.current_line` to zero ? Pressing enter bringing you down to the latest line seems reasonable